dask/optimization.py

```diff
@@ -1023,3 +1023,6 @@ def __call__(self, *args):
 
     def __reduce__(self):
         return (SubgraphCallable, (self.dsk, self.outkey, self.inkeys, self.name))
+
+    def __hash__(self):
+        return hash(self.outkey)
```

Comment:
This is the right idea, but we should also include hashes for `self.inkeys`, `self.name`, and `self.dsk.keys()`  here too

Reply:
Do you mean like the hash of all of these? For example:

```python
hash((self.outkey, tuple(self.dsk.keys()), tuple(self.inkeys), self.name))
```

Reply:
I'd actually drop `dsk.keys()` from the hash here, the outkey should be unique in all cases (and adding the inkeys and name will help). There's bound to be hash collisions in a dict lookup anyway, and the equality check already does a full `self.dsk == other.dsk` check. No need to make the hash call more expensive than it needs to be.